django/db/models/query.py

```diff
@@ -470,6 +470,37 @@ def bulk_create(self, objs, batch_size=None):
 
         return objs
 
+    def bulk_save(self, objs, fields, batch_size=None):
+        """
+        Update the given fields in each of the given objects in the database.
+        Do *not* call save() on each of the instances.
+        """
+        assert batch_size is None or batch_size > 0
+        if not fields:
+            raise ValueError('Specific field names must be given to bulk_save')
+        objs = list(objs)
+        fields = list(fields)
+        if not all(obj.pk for obj in objs):
+            raise ValueError('All objects must have a primary key set')
+        if not objs:
+            return
+        ops = connections[self.db].ops
+        # We use PK twice in the resulting query, once in the CASE statement and once 
+        # in the filter. So we pass in 'pk' twice to account for this.
+        batch_size = (batch_size or max(ops.bulk_batch_size(['pk', 'pk'] + fields, objs), 1))
+        with transaction.atomic(using=self.db, savepoint=False):
+            for batch_objs in [objs[i:i + batch_size] for i in range(0, len(objs), batch_size)]:
+                pks = [obj.pk for obj in batch_objs]
+                update_kwargs = {
+                    field: Case(
+                        *(When(pk=obj.pk, then=Value(getattr(obj, field)))
```

Comment:
Careful - Value only works with basic types that the adapter itself knows how to convert to SQL. You could try passing the fields Field as the `output_field` param to `Value` which will get it to translate the value using the fields get_db_prep_* methods.

Unfortunately Fields are not themselves Expressions, which is why you've chosen to wrap the field in a Value.

We could consider the following options:

1. Implement the Expression API on fields directly
2. Add a FieldExpression that wraps a field and translates the Expressions API onto the Field API

Personally, I've been thinking about 1 being useful a lot. Fields should just be another type of expression. If I were doing fields from scratch, they would be. If just passing the fields Field instance to output_field doesn't work for more complex types, then you might need to consider the options above.

Reply:
Thanks for the feedback! Ok, I will add some tests for more complex fields and try passing them as an output_field. Can you give me some examples of such complesx fields? Do you mean fields related to GIS, or postgres specific ones?

This would indeed be a limitation. Implementing the expression API on fields seems like the cleaner way to do things.

Reply:
I agree with `Implement the Expression API on fields directly`, is there a ticket for that already?

Going off a tangent, but we should keep in mind that not every field inherits from `models.Field` (e.g. `contrib.contenttypes`), I wonder if we should introduce a `BaseField` class where things like expression support would be added.